webapp/src/components/tutorial/TutorialStepCounter.tsx

```diff
@@ -44,6 +44,24 @@ export function TutorialStepCounter(props: TutorialStepCounterProps) {
     const nextButtonTitle = lastStep ? lf("Finish the tutorial.") : lf("Go to the next step of the tutorial.");
     const nextButtonAction = lastStep ? props.onDone : handleNextStep;
 
+    const previousButton = props.isHorizontal ? (
```

Comment:
I don't know if this suggestion would be better/beneficial/more readable, but one thing that can be done instead of making the ternary operator between two button elements would be just to have the ternary be on `className` and `leftIcon` . I notice that you had the ternary on `leftIcon` before, would love to know the context on this change. I know this would make those lines kinda long and clunky and I'm not sure what the performance implications are, either, so take it with a grain of salt.

Reply:
I'm open to either. I actually originally implemented it with the ternary operator on the classname and icon fields, then switched to this because I thought it was a little easier to read/understand/debug if someone is just looking at the code, but it's all a matter of preference and I don't feel strongly. If there's a preference to do it inside the strings instead, I'm open to switching it.

As far as performance, I haven't checked but I would be surprised if there was any measurable impact either way.

Reply:
I have a slight preference for doing it in the strings because there is so much that is similar between the two buttons and I've seen a lot of other components in our code base change class name with ternary operators, so it's a bit more consistent with code style.

Reply:
Yeah, this is not a place where we have to be considerate of performance, it'll never matter (especially in the context of something that would be run once or twice a second -- e.g. it's like 1 second for a billion computations including ternaries :)

```javascript
(() => {
  let sum = 0;
  let start = Date.now()
  for (let i = 0; i < 1000000000; ++i) {
    sum += i % 2 === 0 ? i : 0;
  }
  console.log(`${Date.now() - start}ms to calculate ${sum}`);
})();
```
<img width="321" alt="image" src="https://user-images.githubusercontent.com/5615930/228372794-a179a98c-07a5-483e-a794-31fb7701826a.png">

if you wanna make it so just the icon / classname are set off a ternary it'd probably be just as readable to set them as a variable `backButtonClass` / `backButtonIcon` just above, similar to how `nextButtonTitle` / `nextButtonAction` are currently being set

Reply:
I made the change Sarah recommended here. Seemed simplest to just make it all inline rather than create a bunch of single-use variables, so I went with that approach. I don't love the nested ternaries but we're making compromises either way.